askama/src/lib.rs

```diff
@@ -76,23 +76,50 @@ pub use askama_escape::{Html, Text};
 pub trait Template {
     /// Helper method which allocates a new `String` and renders into it
     fn render(&self) -> Result<String> {
-        let mut buf = String::with_capacity(self.size_hint());
+        let mut buf = String::with_capacity(Self::SIZE_HINT);
         self.render_into(&mut buf)?;
         Ok(buf)
     }
     /// Renders the template to the given `writer` buffer
-    fn render_into(&self, writer: &mut dyn std::fmt::Write) -> Result<()>;
+    fn render_into<W: std::fmt::Write + ?Sized>(&self, writer: &mut W) -> Result<()>;
+
+    /// The template's extension, if provided
+    const EXTENSION: Option<&'static str>;
```

Comment:
Do we have a preference for item order? For instance, in `impl`s, I usually do `type`, then `const`, then all `fn`s. The `type` then `fn` seems to be _standard_ in Rust. For consistency, should be place `const` first, last, anywhere?

Reply:
Oh, interesting. Personally, I order stuff in groups, and then within each group I order by visibility, so `pub` comes first. I recall [this Reddit comment](https://www.reddit.com/r/rust/comments/453ipw/comment/czuwj4z/?context=3), it's 90% of how I order my code, in my own projects.

That said, I don't really care what the order is. It was more a question of, if there was an order, so that I know it going forward. If you prefer `const` at the bottom, then that's fine with me :)